### PR TITLE
Feature/auto workflow install

### PR DESCRIPTION
**Pull request by ilveroluca**
This PR implements automatic workflow installation into our Galaxy instance.

At this stage, workflows to be installed need to be serialized (exported from Galaxy) and saved into the `workflows` directory of this repository (and thus, the resulting container).  When the instance is started the code will take care to install them, unless they're are already present in the instance (identity is discerned via the the workflow's uuid field).

This PR also includes some improvements to the logging and error-reporting for the rest of the Galaxy configuration code.

### CODE REVIEW

File: ansible/configure_galaxy.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                        default='.')
+
+    options = parser.parse_args(args)
+
+    Log.debug("Galaxy config running from %s", os.path.abspath('.'))
+    if len(options.user_password) < 6:
+        parser.error("Password needs to be at least 6 characters long.  User not added!")
+    if not re.match(r'[^@]+@[^@]+\.[^@]+', options.user_email):
+        parser.error("Email address has an invalid syntax.  User not added!")
+    if not os.path.isdir(options.workflow_dir) or not os.access(options.workflow_dir, os.R_OK | os.X_OK):
```
Comment (pcm32) [inline ansible/configure_galaxy.py]: I'm not sure if it is a good idea to tie workflows and users together. I might want to setup users, but not workflows, or the other way around. I think that if the workflow dir option is not passed, then I wouldn't stop the process. I wonder whether these two might actually be two separate scripts altogether (I see the advantage of sharing the connection process though).

Comment (pcm32) [inline ansible/configure_galaxy.py]: Might be cleaner if we have two separate scripts, but I don't oppose having the two functionalities in the same script if it is more convenient.

Comment (ilveroluca) [inline ansible/configure_galaxy.py]: I can change it to set a user and no workflows, but not the other way around I'm afraid.

There it no real user account associated with the "master" api key so some operations can't be performed.  Among these is workflow installation.  This means that  we need to set up the specified user and create an API key for the account to install the workflows.

This same issue makes it a little messy to separate the steps into multiple scripts.  I'd have to output the (potentially new) admin user's API key from the first script (I suppose by writing it to stdout), parse that output and pass it into the second script.  I think it's simpler and more robust to keep all galaxy configuration steps in the same Python script, where it's easier to control how we pass things around.

In conclusion, I'm thinking of resolving this by adding a `--skip-steps=...` option that can accept a list of arguments.  For now the only possible step to skip is workflow installation; in the future more may be possible.  In all cases, user set-up is mandatory.  What do you think?

Comment (pcm32) [inline ansible/configure_galaxy.py]: I understand, thanks for the clear explanation, I think that your proposed approach makes sense given the restrictions. Maybe instead of adding the extra argument for skipping, you could simply avoid failing and not calling the workflow part if the workflow argument is missing, instead of having to actively tell this to skip the step, but whatever you decide there I'm sure it will be fine. Thanks!

Comment (pcm32) [inline ansible/configure_galaxy.py]: In any case, I think that if you are very busy, we can leave this disentanglement for a later feature, there is no immediate need. Feel free to merge if you want.
